### PR TITLE
kinetis/fcfield-check*: merge into single file and support binary flash

### PR DESCRIPTION
**Pull request by fjmolinas**
### Contribution description

This PR merges the two existing scripts to check for the fcfield contents on kinetis boards. It also adds to the merged scipt the capacity of checking for bin files.

Regarding flashing at an offset, since the `fcfield` content is only read on reset, we don't care for its content when flashing an Image after 0x410 or 1040, therefore we won't check its content.

### Testing procedure

Modify any value in the fcfield:

 https://github.com/RIOT-OS/RIOT/blob/8fe12bc82cfbb83a90efd018e11c044d0a40696b/cpu/kinetis/fcfield.c#L55-L73

Flash an elf, hex and bin file and see that it is properly checked:

* `make -C examples/hello-world/ BOARD=frdm-kw41z flash`

```
### Flashing Target ###
Danger of bricking the device during flash!
Flash configuration field of /home/francisco/workspace/RIOT/examples/hello-world/bin/frdm-kw41z/hello-world.elf:

/home/francisco/workspace/RIOT/examples/hello-world/bin/frdm-kw41z/hello-world.elf:     file format elf32-littlearm

Contents of section .fcfield:
 0400 ffffffff ffffffff aaffffff feffffff  ................
Abort flash procedure!
pre-flash checks failed, status=1
```

* `FFLAGS='flash $(HEXFILE)' make -C examples/hello-world/ BOARD=frdm-kw41z flash`

```
### Flashing Target ###
Danger of bricking the device during flash!
Flash configuration field of /home/francisco/workspace/RIOT/examples/hello-world/bin/frdm-kw41z/hello-world.hex:

/home/francisco/workspace/RIOT/examples/hello-world/bin/frdm-kw41z/hello-world.hex:     file format ihex

Contents of section .sec1:
 0400 ffffffff ffffffff aaffffff feffffff  ................
Abort flash procedure!
pre-flash checks failed, status=1
```

* `FFLAGS='flash $(BINFILE)' make -C examples/hello-world/ BOARD=frdm-kw41z binfile flash`

### Flashing Target ###
Danger of bricking the device during flash!
Flash configuration field of /home/francisco/workspace/RIOT/examples/hello-world/bin/frdm-kw41z/hello-world.bin:

```
/home/francisco/workspace/RIOT/examples/hello-world/bin/frdm-kw41z/hello-world.bin:     file format binary

Contents of section .data:
 0400 ffffffff ffffffff aaffffff feffffff  ................
Abort flash procedure!
pre-flash checks failed, status=1

```
Remove the changes to fcfield, and repeat the above procedure, no errors will be encountered.

### Issues/PRs references

Used by #11562

### CODE REVIEW

File: cpu/kinetis/dist/check-fcfield.sh [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    exit 2
+fi
+
+FLASHFILE="$1"
+
+FCFIELD_START='0x400'
+FCFIELD_END='0x410'
+FCFIELD_AWK_REGEX='^ 0400 '
+
+if [ ${FLASHFILE##*.} = bin ]; then
```
Comment (cladmi) [inline cpu/kinetis/dist/check-fcfield.sh]: I would prefer adding the `.bin` and `IMAGE_OFFSET` in a second commit, to show "this was the old behavior moved in the same file`, `this is adding `.bin` support.

File: cpu/kinetis/dist/check-fcfield.sh [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    OBJDUMP=$(arm-none-eabi-objdump --start-address=${FCFIELD_START} --stop-address=${FCFIELD_END} -bbinary -marm ${FLASHFILE} -s)
```
[the remark this answers is not in the document]
Comment (fjmolinas) [reply cpu/kinetis/dist/check-fcfield.sh]: Thanks for the feedback!

**Comment by fjmolinas** [comment]
@cladmi I have addressed your comments. I split merging the scripts and adding binary check in separate commits. Also refactored to use functions and avoid assigning command output to variables, Hopefully it looks better now :) ! (And thanks for the feedback)

**Comment by cladmi** [comment]
I get the correct output without offset:

```
RIOT_CI_BUILD=1 BOARD=frdm-kw41z make -C examples/hello-world/ flash
make: Entering directory '/home/harter/work/git/RIOT/examples/hello-world'
Building application "hello-world" for "frdm-kw41z" with MCU "kinetis".

   text    data     bss     dec     hex filename
   9084     116    2544   11744    2de0 /home/harter/work/git/RIOT/examples/hello-world/bin/frdm-kw41z/hello-world.elf
/home/harter/work/git/RIOT/dist/tools/openocd/openocd.sh flash /home/harter/work/git/RIOT/examples/hello-world/bin/frdm-kw41z/hello-world.elf
### Flashing Target ###
Danger of bricking the device during flash!
Flash configuration field of /home/harter/work/git/RIOT/examples/hello-world/bin/frdm-kw41z/hello-world.elf:

/home/harter/work/git/RIOT/examples/hello-world/bin/frdm-kw41z/hello-world.elf:     file format elf32-littlearm

Contents of section .fcfield:
 0400 00ffffff ffffffff ffffffff feffffff  ................
Abort flash procedure!
pre-flash checks failed, status=1
/home/harter/work/git/RIOT/examples/hello-world/../../Makefile.include:538: recipe for target 'flash' failed
make: *** [flash] Error 1
make: Leaving directory '/home/harter/work/git/RIOT/examples/hello-world'
```

```
FLASHFILE='$(HEXFILE)' RIOT_CI_BUILD=1 BOARD=frdm-kw41z make -C examples/hello-world/ flash
make: Entering directory '/home/harter/work/git/RIOT/examples/hello-world'
Building application "hello-world" for "frdm-kw41z" with MCU "kinetis".

   text    data     bss     dec     hex filename
   9084     116    2544   11744    2de0 /home/harter/work/git/RIOT/examples/hello-world/bin/frdm-kw41z/hello-world.elf
/home/harter/work/git/RIOT/dist/tools/openocd/openocd.sh flash /home/harter/work/git/RIOT/examples/hello-world/bin/frdm-kw41z/hello-world.hex
### Flashing Target ###
Danger of bricking the device during flash!
Flash configuration field of /home/harter/work/git/RIOT/examples/hello-world/bin/frdm-kw41z/hello-world.hex:

/home/harter/work/git/RIOT/examples/hello-world/bin/frdm-kw41z/hello-world.hex:     file format ihex

Contents of section .sec1:
 0400 00ffffff ffffffff ffffffff feffffff  ................
Abort flash procedure!
pre-flash checks failed, status=1
/home/harter/work/git/RIOT/examples/hello-world/../../Makefile.include:538: recipe for target 'flash' failed
make: *** [flash] Error 1
make: Leaving directory '/home/harter/work/git/RIOT/examples/hello-world'
```

```FLASHFILE='$(BINFILE)' RIOT_CI_BUILD=1 BOARD=frdm-kw41z make -C examples/hello-world/ flash
make: Entering directory '/home/harter/work/git/RIOT/examples/hello-world'
Building application "hello-world" for "frdm-kw41z" with MCU "kinetis".

   text    data     bss     dec     hex filename
   9084     116    2544   11744    2de0 /home/harter/work/git/RIOT/examples/hello-world/bin/frdm-kw41z/hello-world.elf
/home/harter/work/git/RIOT/dist/tools/openocd/openocd.sh flash /home/harter/work/git/RIOT/examples/hello-world/bin/frdm-kw41z/hello-world.bin
### Flashing Target ###
Danger of bricking the device during flash!
Flash configuration field of /home/harter/work/git/RIOT/examples/hello-world/bin/frdm-kw41z/hello-world.bin:

/home/harter/work/git/RIOT/examples/hello-world/bin/frdm-kw41z/hello-world.bin:     file format binary

Contents of section .data:
 0400 00ffffff ffffffff ffffffff feffffff  ................
Abort flash procedure!
pre-flash checks failed, status=1
/home/harter/work/git/RIOT/examples/hello-world/../../Makefile.include:538: recipe for target 'flash' failed
make: *** [flash] Error 1
make: Leaving directory '/home/harter/work/git/RIOT/examples/hello-world'
```

File: cpu/kinetis/dist/check-fcfield.sh [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+}
+
+if [ $# -ne 1 ]; then
+    echo "Usage: $0 FLASHFILE"
+    echo "Checks the flash configuration field protection bits to avoid flashing a locked image to a Kinetis MCU (protection against accidental bricking)."
+    exit 2
+fi
+
+if [ $(printf '%d' "${IMAGE_OFFSET}") -gt $(printf '%d' "${FCFIELD_END}") ] ; then
+    echo "WARN: we don't check the fcfield value if flashing at \$IMAGE_OFFSET > 0x410"
```
Comment (cladmi) [inline cpu/kinetis/dist/check-fcfield.sh]: No need to warn here I think. It is a normal case.

**Comment by fjmolinas** [comment]
@cladmi Addressed comments!

**Comment by cladmi** [comment]
Tests from my previous comments now work as I expect.
I will finish the review.

**Comment by cladmi** [comment]
Please squash.

For the commit in `fcfield` you can precise in the comment that it is placed at the address `0x400` in memory, so not relative to the `.vector` position.

**Comment by cladmi** [comment]
I could correctly use this one with #11562 
I am re-running the test suit for good measure.

**Pull request merged**